network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/exclusion/ui/ManuallyDisableAppProtectionDialog.kt

```diff
@@ -0,0 +1,131 @@
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.exclusion.ui
+
+import android.app.Dialog
+import android.os.Bundle
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.fragment.app.DialogFragment
+import com.duckduckgo.app.global.extensions.safeGetApplicationIcon
+import com.duckduckgo.networkprotection.impl.R
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class ManuallyDisableAppProtectionDialog : DialogFragment() {
+
+    interface ManuallyDisableAppProtectionDialogListener {
+        fun onAppProtectionDisabled(
+            appName: String,
+            packageName: String,
+            report: Boolean = false,
+        )
+    }
+
+    val listener: ManuallyDisableAppProtectionDialogListener
+        get() {
+            return if (parentFragment is ManuallyDisableAppProtectionDialogListener) {
+                parentFragment as ManuallyDisableAppProtectionDialogListener
+            } else {
+                activity as ManuallyDisableAppProtectionDialogListener
+            }
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        isCancelable = false
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val rootView = layoutInflater.inflate(R.layout.dialog_netp_manually_disable_app, null)
+
+        val appIcon = rootView.findViewById<ImageView>(R.id.netp_disable_dialog_app_icon)
+        val appLabel = rootView.findViewById<TextView>(R.id.netp_disable_dialog_label)
+        val reportCTA = rootView.findViewById<Button>(R.id.netp_disable_dialog_action_report)
+        val skipCTA = rootView.findViewById<Button>(R.id.netp_disable_dialog_action_skip)
+
+        val alertDialog = MaterialAlertDialogBuilder(requireActivity(), com.duckduckgo.mobile.android.R.style.Widget_DuckDuckGo_Dialog)
+            .setView(rootView)
+
+        validateBundleArguments()
+        isCancelable = false
+
+        populateAppIcon(appIcon)
+        populateAppName(appLabel)
+        configureListeners(reportCTA, skipCTA)
+
+        return alertDialog.create()
+    }
+
+    private fun populateAppIcon(appIcon: ImageView) {
+        val icon = requireActivity().packageManager.safeGetApplicationIcon(getPackageName())
+        appIcon.setImageDrawable(icon)
+    }
+
+    private fun populateAppName(appLabel: TextView) {
+        appLabel.text = getString(R.string.netpExclusionListManuallyDisableAppDialogLabel, getAppName())
+    }
+
+    private fun getPackageName(): String {
+        return requireArguments().getString(KEY_APP_PACKAGE_NAME)!!
+    }
+
+    private fun getAppName(): String {
+        return requireArguments().getString(KEY_APP_NAME)!!
+    }
+
+    private fun configureListeners(
+        reportCTA: Button,
+        skipCTA: Button,
+    ) {
+        reportCTA.setOnClickListener {
```

Comment:
Dunno about your code style but kotlin naming conventions suggests to name reportCta (acronym has 3+ letters)
(c) https://kotlinlang.org/docs/coding-conventions.html#choose-good-names

Reply:
Thanks for the comment @dryaz, we're not precious about that particular convention